src/main/kotlin/com/atlassian/performance/tools/virtualusers/api/VirtualUserOptions.kt

```diff
@@ -405,9 +417,16 @@ class VirtualUserOptions(
                 val browserClass = Class.forName(browser)
                 val browserConstructor = browserClass.getConstructor()
                 (browserConstructor.newInstance() as Browser)::class.java
-            } else {
-                GoogleChrome::class.java
-            }
+            } else GoogleChrome::class.java
+        }
+
+        private fun getLogging(commandLine: CommandLine): Class<out AbstractConfiguration> {
+            return if (commandLine.hasOption(loggingParameter)) {
+                val logging = commandLine.getOptionValue(loggingParameter)
+                val loggingClass = Class.forName(logging)
+                val loggingConstructor = loggingClass.getConstructor()
+                (loggingConstructor.newInstance() as AbstractConfiguration)::class.java
+            } else LogConfiguration::class.java
```

Comment:
If you dislike an uneven `if-else` structure, we can go for:

### Good old extraction
```kotlin
private fun getLogging(commandLine: CommandLine): Class<out AbstractConfiguration> {
    return if (commandLine.hasOption(loggingParameter)) {
        parseLoggingClass(commandLine)
    } else {
        LogConfiguration::class.java
    }
}

private fun parseLoggingClass(commandLine: CommandLine): Class<out AbstractConfiguration> {
    val logging = commandLine.getOptionValue(loggingParameter)
    val loggingClass = Class.forName(logging)
    val loggingConstructor = loggingClass.getConstructor()
    return (loggingConstructor.newInstance() as AbstractConfiguration)::class.java
}
```

### Single expression
```kotlin
private fun getLogging(
    commandLine: CommandLine
): Class<out Configuration> = commandLine
    .getOptionValue(loggingParameter)
    ?.let { Class.forName(it) }
    ?.let { it.getConstructor() }
    ?.let { it.newInstance() as Configuration }
    ?.let { it::class.java }
    ?: LogConfiguration::class.java
```

### Accept it
Or we can just live with a slightly uneven if-else blocks. 8 lines is not much, it's basically a short snippet.

Reply:
Maybe I misunderstand something but does it mean that methods like getScenario and getBrowser will remain as not refactored (good old extraction)?
I like the second approach (Single expression) it looks more like functional code which I found that type of style programming across the repository.

Reply:
You can apply the refactor to nearby code for consistency